src/radical/entk/utils/provenance.py

```diff
@@ -0,0 +1,82 @@
+
+__copyright__ = 'Copyright 2023, The RADICAL-Cybertools Team'
+__license__   = 'MIT'
+
+from typing import Optional, Dict, List
+
+import radical.utils as ru
+
+from .. import Pipeline
+
+
+# ------------------------------------------------------------------------------
+#
+def get_provenance_graph(pipelines: List[Pipeline],
+                         output_file: Optional[str] = None) -> Dict:
+    """
+    Using UIDs of all entities to build a workflow provenance graph.
+    """
+
+    graph = {}
+
+    pipelines = ru.as_list(pipelines)
+
+    for pipeline in pipelines:
+        graph[pipeline.uid] = {}
+
+        for stage in pipelines.stages:
+            graph[pipeline.uid][stage.uid] = []
+
+            for task in stage.tasks:
+                graph[pipeline.uid][stage.uid].append(task.uid)
+
+            graph[pipeline.uid][stage.uid].sort()
+
+    if output_file:
+        if '/' not in output_file:
```

Comment:
Why do you want to enforce a `/` in the file name?

Reply:
thought to be explicit where file is saved. Should I remove this check? (and only to keep `.json` check)

Reply:
Yeah, I would suggest to remove.  Thanks!